### ISSUE TITLE
Compilation Error and Syntax Help

### ISSUE THREAD

**Issue by thomasbaldwin** [Victorma/LeapTrainer.CSharp#1]
Hi @Victorma !

First of all, I want to say thanks for creating this. This is a life saver for a project me and a couple friends are working on for school.

However, I'm getting some issues when trying to setup. For starters, I can't seem to compile because of this error:

![capture-1](https://cloud.githubusercontent.com/assets/5099084/18894417/bc40f7ea-84c7-11e6-992b-e0ff5f024ad1.PNG)

Clicking into it reveals this:

![capture-2](https://cloud.githubusercontent.com/assets/5099084/18894422/c456c7ca-84c7-11e6-8abe-18b247c6ff0f.PNG)

Any ideas what I might be doing wrong?

Lastly, I'm trying to execute this code from the original [JS LeapTrainer](https://github.com/roboleary/LeapTrainer.js), but I can't seem to figure out which syntax to use. Here is the syntax from the original [JS LeapTrainer](https://github.com/roboleary/LeapTrainer.js):

``` javascript
var trainer = new LeapTrainer.Controller();
trainer.create('Halt', false); // The controller will switch to training mode
trainer.on('Halt', function() { console.log('Stop right there!'); });
```

And here's where I'm currently at in C#:

``` c#
using UnityEngine;
using System.Collections;

public class ThomasLeapTrainer : MonoBehaviour {
    // Use this for initialization
    void Start () {
        LeapTrainer trainer = new LeapTrainer();

        trainer.Create("Halt", false);
        // not sure what I should be passing in here, if this even is the correct method to be calling
        trainer.OnGestureDetected()
    }

    // Update is called once per frame
    void Update () {

    }
}
```

Thanks for your help!
Thomas

**Comment by Victorma** [comment]
Hi @thomasbaldwin !

First of all, you can check my project "SignAdventure" where I use this library. Specially you can check this class to see how I make it work: https://github.com/Victorma/SignAdventure/blob/master/Assets/Play.cs#L39 

Second, I am pretty sure that your issue is coming from the Leap Motion software you're using (I am not 100% sure but if I recall correctly, when I updated the code to the last version it changed the way the listener works).

So make sure you're using **Orion** version of Leap Motion SDK. You can get it here: https://developer.leapmotion.com/get-started
And the Unity plugins:
https://developer.leapmotion.com/unity

Please, try it and if it still doesn't work or you have any doubts, tell me back!

**Comment by Victorma** [comment]
Also, I whould like to mention that I dont have a visual interface to train the recognizer, so the way I train it, is using this method:
https://github.com/Victorma/SignAdventure/blob/master/Assets/Play.cs#L133

All you have to do is previously record the frames using the tool provided by Leap Motion Examples.
Mine is a little bit changed in order to add a Delay after I press the letter R and start recording.
https://github.com/Victorma/SignAdventure/blob/master/Assets/LeapMotion/Scenes/RecordAndPlayback.unity

Then I take the .asset file saved into the folder written in the console and use it inside my code.
I think also the "Tutorial" class when it loads, it discards some frames in the beginning and in the end.
https://github.com/Victorma/SignAdventure/blob/master/Assets/Tutorial.cs#L81

Also if I recall correctly I changed the code inside https://github.com/Victorma/SignAdventure/blob/master/Assets/LeapMotion/Scripts/Utils/LeapRecorder.cs to do something I dont remember now, like retrieve frames or add new frames to them... Sorry...

Good Luck!

**Comment by bridgewaterrobbie** [comment]
@Victorma 
Hey, Im with @thomasbaldwin 

We are using Orion, though there seem to have been some changes between the current version of Orion and what we had.

However, I dont think that you used Orion in Sign Adventure, at least not the version you have posted to Github. I say this based on two things: First is that the sign adventure project you have posted has a version of Leap that accepts a Tool Model, which I think was depracated in Orion.

Second is that the readme here: https://github.com/Victorma/SignAdventure/tree/master/Assets/LeapMotion says "These assets go along with our V2 Skeletal Tracking"

The Listener in Sign Adventure points to a folder called LeapMotionVR/SystemWipe , which doesn't seem to have a corresponding folder in any Orion version I have found. 

From your last post though, it sounds like you already tackled this! If you could update the version of LeapTrainer.CSharp you have on github, we would be thankful. Otherwise we will go ahead and try to do that.

Thank you so much for doing this and open sourcing it though. It takes care of a huge implementation step on our research project and we really appreciate it.

**Comment by Victorma** [comment]
Hi @thomasbaldwin ( or @bridgewaterrobbie )
As I was reading your post, I started to think about it and you're probably right indeed.

I remember I wanted to use realistic hands in the project and by the time I was doing Sign Adventure, only skeleton robotic hands were avaliable so I decided to go back to the other software.

Anyway, I'll take a look to the code asap just in case I find a way to solve this!

Also, I'll try to upload the latest custom changes to this repo. 

I'm answering this as fast as possible because I'm pretty busy this week, but I'll have more time in the weekend, so you know I'm already aware of this. Thanks to you for using my code!

**Comment by thomasbaldwin** [comment]
Thanks @Victorma !!

**Comment by thomasbaldwin** [comment]
Hey @Victorma ,

Been playing around with your code for the last couple of days, but haven't quite got things working with v3 yet. Have you been you able to make any progress?

Thanks for your time,
Thomas

**Comment by Victorma** [comment]
Hi @thomasbaldwin !
Not really to be honest, I had no time but I'm going to take a look to it tomorrow for sure ;)
Thanks!

**Comment by Victorma** [comment]
Hey @thomasbaldwin !
I've been updating my Sign Adventure repo in order to make it work with the new Orion version and Apparently I succeed. I'll update the few little changes here but I have one very big issue I cant solve yet.

Since LeapRecorder has been removed from the Orion bundle, I had to copy it from previous release. It is working nicely, and I added a new LeapServiceProvider in order to use the LeapRecorder and provide the frames to the new LeapHandController.

Sadly, frame deserialization is not working (probably because the frame has ben changed between V2 and Orion) and I am not able to test it because I don't have the LeapMotion anymore.

The changes inside LeapTrainer are mainly:
- This: https://github.com/Victorma/SignAdventure/blob/a263d4d89a87a382ece2b7af8db005750da701d3/Assets/LeapClasifier/LeapTrainer.cs#L16
- And this: https://github.com/Victorma/SignAdventure/blob/a263d4d89a87a382ece2b7af8db005750da701d3/Assets/LeapClasifier/LeapTrainer.cs#L131

The LeapRecorder and RecordedServiceProvider are here:
https://github.com/Victorma/SignAdventure/tree/orion/Assets/LeapMotionModules/Utils

The way the trainer is teached is the same as before:

``` chsarp
recorder.Load(asset);
trainer.loadFromFrames (name, recorder.GetFrames(), false);
```

I'll try to update this repo later in a new branch also.

**Comment by bridgewaterrobbie** [comment]
@Victorma 
Thank you so much for your help. We are figuring out how to use it now in full, but I think we got it.

One minor note: The changes cause some bugs in sign adventure. It doesn't bother us (If you touch start with the leap it crashes, and if you try to do the training mode it crashes) but I thought you might want to know.

Do we have your permission to publicly include this code as part of an augmented reality development setup? We would give you credit of course.

**Comment by Victorma** [comment]
@bridgewaterrobbie Yes of course! I took it from @roboleary and both versions of LeapTrainer are licensed under MIT, so feel free to use it as you want! 

Dont worry about Sign Adventure, I know the errors, but since I dont have the Leap Motion anymore is a lost cause for now ;) Thanks anyway

Happy Hacking!
